pkg/config/event.go

```diff
@@ -74,30 +75,81 @@ func readEnv(key string, def string) string {
 	return value
 }
 
+// SetupCloudEventsKafkaClient create the cloud events kafka client that will send events to the given kafka topic
+func SetupCloudEventsKafkaClient(topic string) (v2.Client, error) {
+	kafkaServers := strings.Split(LoadedConfig.Kafka.Bootstrap.Servers, ",")
+	saramaConfig, err := GetSaramaConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error getting sarama config: %w", err)
+	}
+
+	topicTranslator := event.NewTopicTranslationWithClowder(clowder.LoadedConfig)
+	mappedTopicName := topicTranslator.GetReal(topic)
+
+	if mappedTopicName == "" {
+		mappedTopicName = topic
+	}
+
+	protocol, err := kafka_sarama.NewSender(kafkaServers, saramaConfig, mappedTopicName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kafka_sarama protocol: %w", err)
+	}
+
+	c, err := v2.NewClient(protocol, v2.WithTimeNow(), v2.WithUUIDs())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create cloud events client: %w", err)
+	}
+	return c, nil
+}
+
+// SetupTemplatesNotifications creates the cloud events kafka client for sending events to the patch service
+func SetupTemplatesNotifications() {
+	if !LoadedConfig.Options.EnableTemplatesNotifications {
+		return
+	}
+
+	if len(LoadedConfig.Kafka.Bootstrap.Servers) == 0 {
+		log.Warn().Msg("SetupTemplatesNotifications: clowder.KafkaServers and configured broker was empty")
+		return
+	}
+
+	client, err := SetupCloudEventsKafkaClient("platform.content-sources.template")
+	if err != nil {
+		log.Error().Err(err).Msg("SetupTemplatesNotifications failed")
+		return
+	}
+	LoadedConfig.TemplatesNotificationsClient = client
+}
+
+// SetupNotifications creates the cloud events kafka client for sending events to the notifications service
 func SetupNotifications() {
 	if !LoadedConfig.Options.EnableNotifications {
 		return
 	}
 
 	if len(LoadedConfig.Kafka.Bootstrap.Servers) == 0 {
 		log.Warn().Msg("SetupNotifications: clowder.KafkaServers and configured broker was empty")
+		return
 	}
 
-	kafkaServers := strings.Split(LoadedConfig.Kafka.Bootstrap.Servers, ",")
+	client, err := SetupCloudEventsKafkaClient("platform.notifications.ingress")
+	if err != nil {
+		log.Error().Err(err).Msg("SetupNotifications failed")
+		return
+	}
+	LoadedConfig.NotificationsClient = client
+}
+
+func GetSaramaConfig() (*sarama.Config, error) {
 	saramaConfig := sarama.NewConfig()
 
 	saramaConfig.Version = sarama.V2_0_0_0
 	saramaConfig.Consumer.Offsets.Initial = sarama.OffsetOldest
 
-	if strings.Contains(LoadedConfig.Kafka.Sasl.Protocol, "SSL") {
-		saramaConfig.Net.TLS.Enable = true
-	}
```

Comment:
was this unnecessary?

Reply:
good catch

Reply:
I think I just missed it copying/pasting